editor/src/editor/GUIComponents.cpp

```diff
@@ -427,3 +427,21 @@ void SValueMenu::onItemClicked(int32_t index)
     if (getValue() != oldValue)
         valueChanged();
 }
+
+void CHoverButton::setHoverColor (const CColor& color)
+{
+    hoverColor_ = color;
+}
+
+CMouseEventResult CHoverButton::onMouseEntered (CPoint& where, const CButtonState& buttons)
+{
+    backupColor_ = getTextColor();
+    setTextColor(hoverColor_);
+    return CTextButton::onMouseEntered(where, buttons);
+}
```

Comment:
There should be a cleanest way that you can cheat it, if you prefer to avoid messing with color values.
- `setHoverColor` and `onMouse*` set the dirty flag on the control
- override `CTextButton`'s draw method
```
if (hovered) {
    oldTextColor = textColor;
    textColor = myHoveredColor; // textColor is protected
}
CTextButton::draw;
if (hovered)
    textColor = oldTextColor;
```

Also, note that I went with the `S` prefix convention to distinguish these parts from vstgui.

Reply:
Hmm is it so much simpler? 

> Also, note that I went with the `S` prefix convention to distinguish these parts from vstgui.

OK! Will change it later.

Reply:
Not really but it's for cleaner API.
If I'm doing `setTextColor(green)`, I would not expect that `getTextColor()` returns red depending if the mouse is over the part or isn't.
Regarding this button, I'd be tempted renaming it as `STextButton` (a `CTextButton` with extra features)

Reply:
That's a good point.